src/common_swagger_api/schema/badges.clj

```diff
@@ -0,0 +1,29 @@
+(ns common-swagger-api.schema.badges
+  (:use [common-swagger-api.schema :only [describe]])
+  (:require [schema.core :as s])
+  (:import [java.util UUID]))
+
+(def SubmissionField (describe String "The JSON-encoded submission"))
+
+(s/defschema Submission
+  {:id
+   (describe UUID "The UUID for this submission")
+
+   :submission
+   SubmissionField})
+
+(s/defschema NewSubmission
+  (dissoc Submission :id))
+
+(s/defschema Badge
+  {:id
+   (describe UUID "The UUID for the badge")
+
+   :user
+   (describe String "The username of the user that owns the object")
```

Comment:
This is completely optional, but you could define the user field to be of type `NonBlankString`, which is defined in `common-swagger-api.shcema`. Then if someone passed in an empty username, the schema validation would catch it for you. Of course, the username has to be matched against the users table anyway, so this extra level of validation isn't strictly necessary.

Reply:
This should be done as well.